File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;


string solution(string number, int k) {
    int cnt = 0;
    
    while(cnt < k){
        for(int i = 0; i < number.size(); i++){
            if(i == number.size() - 1 || number[i] < number[i + 1]) {
                number.erase(i, 1);
                cnt++;
                break;
            }
        }  
    }
   
    return number;
}

File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.java
import java.util.*;
class Solution {
    public String solution(String number, int k) {
        String answer = "";
        int len = number.length(), cnt = 0;
        
        for(char x : number.toCharArray()) {
            if(answer.equals("")) {
                answer += (x + "");
                continue;
            }
            int cur = Integer.parseInt(x + "");
           
            while(answer.length() > 0){
                int prev = Integer.parseInt(answer.charAt(answer.length() - 1) + "");
            
                if(prev < cur && cnt < k) {
                    answer = answer.substring(0, answer.length() - 1);
                    cnt++;
                } else break;
            }
             answer += (x + "");
        }

        if(cnt < k) answer = answer.substring(0, answer.length() - (k - cnt));
        
        return answer;
    }
}